tests/util.js

```diff
@@ -200,5 +203,141 @@ export default {
 				}
 			],
 		},
+		{
+			name: "isRegisteredProperty()",
+
+			beforeEach () {
+				let iframe = document.createElement("iframe");
+				document.body.append(iframe);
+
+				this.data.iframe = iframe;
+			},
+
+			run (property) {
+				if (typeof property === "object") {
+					property = "--" + property.id;
+				}
+
+				return isRegisteredProperty(property, this.data.iframe.contentWindow);
+			},
+
+			afterEach () {
+				this.data.iframe.remove();
+			},
+
+			getName () {
+				let suffix = "";
+				if (typeof this.arg === "object") {
+					suffix = " (--" + this.arg.id + ")";
+				}
+
+				return this.parent.name + suffix;
+			},
+
+			expect: true,
+
+			tests: [
+				{
+					name: "Inline style",
+					beforeEach () {
+						this.parent.parent.beforeEach.call(this);
+						let property = this.arg;
+						this.data.iframe.contentDocument.head.insertAdjacentHTML(
+							"beforeend",
+							`<style>
+								@property --${property.id} {
+									syntax: "${property.syntax}";
+									inherits: ${property.inherits ?? "true"};
+									initial-value: ${property.initialValue};
+								}
+							</style>`,
+						);
+					},
+					tests: [
+						{
+							arg: types.any,
+							expect: false,
+						},
+						{
+							arg: { ...types.any, id: types.any.id + "-non-inherited", inherits: false },
+						},
+						{
+							arg: types.number,
+						},
+					],
+				},
+				{
+					name: "Adopted stylesheet",
+					beforeEach () {
+						this.parent.parent.beforeEach.call(this);
+						let property = this.arg;
+						adoptCss(`@property --${property.id} {
+							syntax: "${property.syntax}";
+							inherits: ${property.inherits ?? "true"};
+							initial-value: ${property.initialValue};
+						}`, this.data.iframe.contentDocument);
+					},
+					tests: [
+						{
+							arg: types.any,
+							expect: false,
+						},
+						{
+							arg: { ...types.any, id: types.any.id + "-non-inherited", inherits: false },
+						},
+						{
+							arg: types.number,
+						},
+					],
+				},
+				{
+					name: "External stylesheet",
+					async beforeEach () {
+						this.parent.parent.beforeEach.call(this);
+						this.data.iframe.contentDocument.head.insertAdjacentHTML("beforeend", '<link rel="stylesheet" href="./util/properties.css" />');
```

Comment:
Ah, I hadn't noticed that we need a reference to the element after. I think the code you had was actually better in that case, sorry about that!